python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,553 @@
+        gnn_embed_dim : int
+            Size of GNN embeddings.
+        data_dir : str
+            Directory for distillation data.
+        batch_size : int
+            Batch size for distillation.
+        distill_lr : float
+            Learning rate for distillation.
+        saved_path : str
+            Path to save the model.
+        on_cpu : bool
+            Whether the distillation will be conducted on cpu.
+        """
+        self.student = GSDistilledModel(tsf_name, node_type, gnn_embed_dim, pre_trained_name=self.config.pre_trained_name)
+        dataloader_generator = DataloaderGenerator(tokenizer=self.student.tokenizer, 
+            device=self.device, 
+            batch_size=batch_size,
+        )
+        train_data_provider = DataProvider(
+            dataloader_generator,
+            dataset_path=os.path.join(data_dir, 'train'),
+            local_rank=self.rank,
+            world_size=th.distributed.get_world_size(),
+            is_train=True,
+        )
+        eval_data_provider = DataProvider(
+            dataloader_generator,
+            dataset_path=os.path.join(data_dir, 'val'),
+            local_rank=self.rank,
+            world_size=th.distributed.get_world_size(),
+            is_train=False,
+        )
+
+        # TODO (HZ): add flexibility to specify different optimizers
+        optimizer = th.optim.Adam(self.student.parameters(), lr=distill_lr)
+        self.student.to(self.device)
+        student = DistributedDataParallel(self.student, device_ids=None if on_cpu else [self.device],
+                                        output_device=None if on_cpu else self.device)
+
+        index = 0
+        global_step = 0
+        while True:
+            th.distributed.barrier()
+            print (f"Train {index + 1}-th shard by trainer {self.rank}")
+            complete, global_step = self.train_shard(
+                global_step=global_step,
+                model=student, 
+                optimizer=optimizer, 
+                train_dataset_provider=train_data_provider, 
+                eval_dataset_provider=eval_data_provider,
+                saved_path=saved_path,
+            )
+            is_first_iteration = False
+
+            index += 1
+            if complete:
+                break
+
+    def save_student_model(self, model, saved_path, global_step):
+        """ Save student model
+
+        Parameters
+        ----------
+        model : GSDistilledModel
+            Distilled student model.
+        saved_path : str
+            Path to save the model.
+        global_step : int
+            Global step of the model checkpoint.
+        """
+        th.distributed.barrier()
+        if self.rank == 0:
+            os.makedirs(os.path.join(saved_path, model.module.node_type, f"checkpoint-{global_step}"), exist_ok=True)
+            config_file_loc = os.path.join(saved_path, model.module.node_type, f"checkpoint-{global_step}", "config.yaml")
```

Comment:
it seems the config file here is the yaml file. can we copy the yaml file to the directory directly?